plugin/src/main/java/io/jenkins/plugins/casc/core/LabelAtomConfigurator.java

```diff
@@ -0,0 +1,47 @@
+package io.jenkins.plugins.casc.core;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.model.Label;
+import hudson.model.labels.LabelAtom;
+import hudson.model.labels.LabelAtomProperty;
+import io.jenkins.plugins.casc.Attribute;
+import io.jenkins.plugins.casc.BaseConfigurator;
+import io.jenkins.plugins.casc.ConfigurationContext;
+import io.jenkins.plugins.casc.ConfiguratorException;
+import io.jenkins.plugins.casc.impl.attributes.DescribableListAttribute;
+import io.jenkins.plugins.casc.model.Mapping;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import static io.jenkins.plugins.casc.Attribute.noop;
+
+@Extension
+@Restricted(NoExternalUse.class)
+public class LabelAtomConfigurator extends BaseConfigurator<LabelAtom> {
+
+    @Override
+    public Class<LabelAtom> getTarget() {
+        return LabelAtom.class;
+    }
+
+    @Override
+    protected LabelAtom instance(Mapping mapping, ConfigurationContext context) throws ConfiguratorException {
+        return new LabelAtom(mapping.getScalarValue("name"));
+    }
+
+    @NonNull
+    @Override
+    public Set<Attribute<LabelAtom, ?>> describe() {
+        return new HashSet<>(Arrays.asList(
+            new Attribute<LabelAtom, String>("name", String.class)
+                .getter(Label::getName)
+                .setter( noop() ),
+            new DescribableListAttribute<LabelAtom, LabelAtomProperty>("properties", LabelAtomProperty.class)
```

Comment:
`LabelAtomProperty` is not describable so you need a configurator for it too.

Reply:
The class is abstract with no descriptior definition.
https://github.com/jenkinsci/jenkins/blob/6baeb65e2c2cec8e2a22b4fe4e71c803c7373e85/core/src/main/java/hudson/model/labels/LabelAtomPropertyDescriptor.java#L43

That is sure to cause trouble.

Reply:
Isn't it if it extends AbstractDescribableImpl?

Reply:
It is designed that way for plugins to contribute with label properties.